cnf-tests/testsuites/e2esuite/nmstate/nmstate_sriov.go

```diff
@@ -0,0 +1,207 @@
+					DesiredState: nmstateshared.State{
+						Raw: nmstateshared.RawState(fmt.Sprintf(nmstateDesiredStateTemplateStaticIP, testDevice.Name)),
+					},
+				},
+			}
+			err = client.Client.Create(context.Background(), &nodeNetworkConfigPolicy)
+			Expect(err).ToNot(HaveOccurred())
+			DeferCleanup(client.Client.Delete, context.Background(), &nodeNetworkConfigPolicy)
+
+			waitForNMStatePolicyToBeStable(nodeNetworkConfigPolicy.Name)
+
+			By("Configuring VF on device via SriovNetworkNodePolicy")
+			sriovNetworkNodePolicy, err := sriovnetwork.CreateSriovPolicy(sriovclient, "test-nmstate-", namespaces.SRIOVOperator, testDevice.Name, node, 10, "testnmstateresource", "netdevice")
+			Expect(err).ToNot(HaveOccurred())
+			DeferCleanup(sriovclient.Delete, context.Background(), sriovNetworkNodePolicy)
+
+			Eventually(func() int64 {
+				testedNode, err := sriovclient.Nodes().Get(context.Background(), node, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				resNum := testedNode.Status.Allocatable[corev1.ResourceName("openshift.io/testnmstateresource")]
+				capacity, _ := resNum.AsInt64()
+				return capacity
+			}).
+				WithPolling(5*time.Second).
+				WithTimeout(10*time.Minute).
+				Should(Equal(int64(10)), "the number of the observed VFs is not that same as the one sriov created")
+		})
+	})
+})
+
+// findNonPrimarySriovDevicesAndNode returns a node name with its SR-IOV devices. The device list does not include
+// the one used for primary networking, if it is an SR-IOV device.
+// The node with the highest number of SR-IOV devices is returned.
+func findNonPrimarySriovDevicesAndNode(n *cluster.EnabledNodes) (string, []*sriovv1.InterfaceExt, error) {
+	var returnError error
+
+	nodeWithMostDevices := ""
+	deviceList := []*sriovv1.InterfaceExt{}
+
+	for _, node := range n.Nodes {
+		devices, err := n.FindSriovDevices(node)
+		if err != nil {
+			returnError = errors.Join(returnError, err)
+			continue
+		}
+
+		mainDeviceName, err := getPrimaryNICForNode(node)
+		if err != nil {
+			returnError = errors.Join(returnError, err)
+			continue
+		}
+
+		nonPrimaryDevices := []*sriovv1.InterfaceExt{}
+		for _, device := range devices {
+			if device.Name == mainDeviceName {
+				continue
+			}
+			nonPrimaryDevices = append(nonPrimaryDevices, device)
+		}
+
+		if len(nonPrimaryDevices) > len(deviceList) {
+			nodeWithMostDevices = node
+			deviceList = nonPrimaryDevices
+		}
+	}
+
+	if len(deviceList) == 0 {
+		return "", nil, fmt.Errorf("can't find any SR-IOV devices in cluster's nodes: %w", returnError)
+	}
+
+	return nodeWithMostDevices, deviceList, nil
+}
+
+func getPrimaryNICForNode(node string) (string, error) {
+	nodeObj, err := client.Client.Nodes().Get(context.Background(), node, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	buf, err := nodes.ExecCommandOnMachineConfigDaemon(client.Client, nodeObj,
+		[]string{"/bin/bash", "-c", "chroot /rootfs /usr/bin/ovs-vsctl list-ports br-ex | /usr/bin/grep -v patch"})
```

Comment:
probably you can get away with checking interfaces which do not have br-ex as master, not sure it would be less fragile

Reply:
`ip link` on my dev worker node says `master ovs-system`, but it's not unique:

```
$ oc debug node/cnfdr22.telco5g.eng.rdu2.redhat.com -- chroot /host ip link | grep master
4: eno12399: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq master ovs-system state UP mode DEFAULT group default qlen 1000
10: genev_sys_6081: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 65000 qdisc noqueue master ovs-system state UNKNOWN mode DEFAULT group default qlen 1000
14: 39d9e5dfde97b52@if2: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1400 qdisc noqueue master ovs-system state UP mode DEFAULT group default 
16: f50c6bcd4922aa0@if2: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1400 qdisc noqueue master ovs-system state UP mode DEFAULT group default 
17: a9c644ea1db880d@if2: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1400 qdisc noqueue master ovs-system state UP mode DEFAULT group default 
...
```

while `ovs-vsctl` appears to be a little bit more accurate:
```
$ oc debug node/cnfdr22.telco5g.eng.rdu2.redhat.com -- chroot /host /usr/bin/ovs-vsctl list-ports br-ex | /usr/bin/grep -v patch
eno12399
```

I'd go with the ovs-vsctl way, and I'll consider that if I see flakes, I have plan B. WDYT?